packages/tenant-process/src/services/tenantService.ts

```diff
@@ -556,6 +556,100 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async internalAssignCertifiedAttribute(
+      tenantOrigin: string,
+      tenantExternalId: string,
+      attributeOrigin: string,
+      attributeExternalId: string,
+      correlationId: string
+    ): Promise<void> {
+      logger.info(
+        `Assigning certified attribute (${attributeOrigin}/${attributeExternalId}) to tenant (${tenantOrigin}/${tenantExternalId})`
+      );
+
+      const tenantToModify = await this.getTenantByExternalId({
+        origin: tenantOrigin,
+        value: tenantExternalId,
+      });
+
+      assertTenantExists(
+        unsafeBrandId(`${tenantOrigin}/${tenantExternalId}`),
+        tenantToModify
+      );
+
+      const attributeToAssign =
+        await readModelService.getAttributeByOriginAndCode({
+          origin: attributeOrigin,
+          code: attributeExternalId,
+        });
+
+      if (!attributeToAssign) {
+        throw attributeNotFound(`${attributeOrigin}/${attributeExternalId}`);
+      }
+
+      const maybeAttribute = tenantToModify.data.attributes.find(
+        (attr) =>
+          attr.type === tenantAttributeType.CERTIFIED &&
+          attr.id === attributeToAssign.id
+      ) as CertifiedTenantAttribute;
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant: Tenant = {
+        ...tenantToModify.data,
+        updatedAt: new Date(),
+      };
+
+      if (!maybeAttribute) {
+        // assigning attribute for the first time
+        updatedTenant = {
+          ...updatedTenant,
+          attributes: [
+            ...tenantToModify.data.attributes,
+            {
+              id: attributeToAssign.id,
+              type: tenantAttributeType.CERTIFIED,
+              assignmentTimestamp: new Date(),
+              revocationTimestamp: undefined,
+            },
+          ],
+        };
```

Comment:
Same as the point made in https://github.com/pagopa/interop-be-monorepo/pull/381, IMHO there should be a check for the certified attribute existence in the read model